firefox-ios/Client/Frontend/Home/TopSites/DataManagement/TopSitesDataAdaptor.swift

```diff
@@ -124,13 +127,25 @@ class TopSitesDataAdaptorImplementation: TopSitesDataAdaptor, FeatureFlaggable {
         guard shouldLoadSponsoredTiles else { return }
 
         dispatchGroup.enter()
-        contileProvider.fetchContiles { [weak self] result in
-            if case .success(let contiles) = result {
-                self?.contiles = contiles
-            } else {
-                self?.contiles = []
+
+        if featureFlags.isFeatureEnabled(.unifiedAds, checking: .buildOnly) {
+            unifiedAdsProvider.fetchTiles { [weak self] result in
+                if case .success(let unifiedTiles) = result {
+                    self?.contiles = UnifiedAdsConverter().convert(unifiedTiles: unifiedTiles)
```

Comment:
If we go the static route
```suggestion
                    self?.contiles = UnifiedAdsConverter.convert(unifiedTiles: unifiedTiles)
```